Formula/open-mpi.rb

```diff
@@ -39,6 +39,11 @@ class OpenMpi < Formula
   conflicts_with "mpich", because: "both install MPI compiler wrappers"
 
   def install
+    on_linux do
+      # Fix libopen-pal.so: undefined reference to `__atomic_compare_exchange_16'
+      ENV["HOMEBREW_OPTFLAGS"] = "" if build.bottle?
```

Comment:
We try to avoid `build.bottle?` as much as possible. Is this really necessary?

Reply:
I copied this over from the Linux formula but I don't know if it's needed.  Should I reopen the PR in linuxbrew-core and see if it builds in linux CI after I delete that block?  Everything builds and passes tests in my local Linux system.  Perhaps this is leftover from a problem which no longer exists.

Reply:
Quite likely. Definitely drop this line if it builds cleanly without it.

Reply:
Indeed, this builds cleanly and passes tests without it so I'm removing it.